### PR TITLE
doc(README): add table of contents, cleanup title headers and reorg

### PR DESCRIPTION
**Pull request by Alizter**
Rendered
https://github.com/Alizter/coq-lsp/blob/ps/rr/doc_readme___add_table_of_contents__cleanup_title_headers_and_reorg/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,7 +171,37 @@ to interact with Coq.
...
+`coq-lsp` can't work with more than one file at the same time, due to problems
+with parsing state management upstream. This was fixed in Coq `master` branch
+(to become **Coq 8.18**).
+
+As this is very inconvenient, we do provide a fixed Coq branch that you can
+install using `opam pin`:
+
+- For Coq 8.17:
+  ```opam pin add coq-lsp https://github.com/ejgallego/coq.git#v8.17+lsp`
+- For Coq 8.16: `opam pin add coq-lsp https://github.com/ejgallego/coq.git#v8.16+lsp`
```
Comment (ejgallego) [inline README.md:202]: Seems this should be with 3 backticks?

**Review by ejgallego — APPROVED** [review]
thanks a lot!

File: README.md [hunk 2]
```diff
@@ -15,12 +15,42 @@ information panel, performance data, and more.
 experience, as well as to serve as a platform for research and UI integration
 with other projects.
 
-### Installation
+## Table of Contents
+
+- [Coq LSP](#coq-lsp)
+  - [Table of Contents](#table-of-contents)
+  - [Installation](#installation)
```
Comment (ejgallego) [inline README.md:22]: Arent' the TOC itself and the topleve Coq LSP redundant?

File: README.md (hunk 1 above)
Comment (Alizter) [reply README.md:202]: Huh my local changes didn't get pushed, let me fix that.

File: README.md (hunk 2 above)
Comment (Alizter) [reply README.md:22]: My editor automatically generates the TOC, so rather than fighting it I've let it stay.

**Pull request merged**